meson.build

```diff
@@ -1273,6 +1273,18 @@ else
 endif
 conf.set10('HAVE_KMOD', have)
 
+want_xenctrl = get_option('xenctrl')
+if want_xenctrl != 'false' and not skip_deps
+        libxenctrl = dependency('xencontrol',
+                             version : '>= 4.9',
+                             required : want_xenctrl == 'true')
+        have = libxenctrl.found()
+else
+        have = false
+        libxenctrl = []
+endif
+conf.set10('HAVE_XENCTRL', have)
```

Comment:
Please use meson features for new options added in. See https://github.com/systemd/systemd/pull/24175 for how this is supposed to be done.

Reply:
It's too soon for that, it breaks backward compatibility, so for now this is fine as-is

Reply:
It does not. meson features are fine for *new* options we add as we don't have to adhere to compat for those. The minimum meson version we use already supports features. It's only the compat mapping of old/custom options to features which requires newer meson version.

Reply:
Could you get to agree? :)

Reply:
leave as-is, an unrelated PR is not the right place to bikeshed on the style of the meson files

Reply:
> leave as-is, an unrelated PR is not the right place to bikeshed on the style of the meson files

The fuck? This introduces a new feature and thus should do this properly. Old cold being old code is fine. But new code should adhere and use new paradigms of the language. meson features are far superior for this as they make dealing with dependencies far easier, with less code.

Reply:
Even if it is, and I'm not sure that's the case as having 2 ways to do the same thing might get confusing and might not be worth it, again a PR fixing an actual bug on a platform from a first-time contributor is _not_ the right place to bikeshed on something like this. Instead please open a follow-up PR after this is merged, so that it can be discussed without involving @sthibaul who has no skin in this particular game and should not be asked to spend time on it.

Reply:
This isn't bikeshedding. It is requesting to use the *proper meson interface* do add a *new library* instead of using the custom ad-hoc method practiced until then. And it's not a hard burden to do it either, it literally makes the meson code smaller and idiomatic!

As I see it, we practice this all the time: old code can stay using old/deprecated interfaces/processes, but we expect new code to adhere to the new/proper/idiomatic way things are done. Why should our build system be exempt here?

Reply:
Btw, this is what I was asking for (see how it's smaller, easier to follow, idiomatic and no burden to use?):
```meson
libxenctrl = dependency('xencontrol',
                      version : '>= 4.9',
                      required : get_option('xencontrol'))
conf.set10('HAVE_XENCTRL', libxenctrl.found())
```

```
option('xenctrl', type : 'feature',
       description : 'support for Xen kexec')
```